tsdb/chunkenc/xor.go

```diff
@@ -79,14 +79,20 @@ func (c *XORChunk) NumSamples() int {
 	return int(binary.BigEndian.Uint16(c.Bytes()))
 }
 
+// Compact compacts internal buffer to minimum capacity.
 func (c *XORChunk) Compact() {
 	if l := len(c.b.stream); cap(c.b.stream) > l+chunkCompactCapacityThreshold {
-		buf := make([]byte, l)
-		copy(buf, c.b.stream)
-		c.b.stream = buf
+		c.b.stream = c.Clone().Bytes()
 	}
 }
 
+// Clone return exact, compacted copy of chunk.
+func (c *XORChunk) Clone() Chunk {
```

Comment:
If this is still needed, it should have a unittest.

Reply:
ack let's do it!